### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt [hunk 1]
```diff
@@ -127,7 +127,7 @@ class ClearlyDefinedStorage(
         try {
             readFromClearlyDefined(id, packageCoordinates(id, vcs, sourceArtifact))
         } catch (e: IllegalArgumentException) {
-            log.warn("Could not obtain ClearlyDefined coordinates for package '${id.toCoordinates()}'.", e)
+            log.warn { "Could not obtain ClearlyDefined coordinates for package '${id.toCoordinates()}'." }
```
Comment (mnonnenmacher) [inline scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt:130]: You should also add `e.showStackTrace()`, especially as you mention `--stacktrace` in the commit message. Or you could add ": $e.collectMessagesAsString()" as we do in other places. Or both.

File: scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt [hunk 2]
```diff
@@ -158,7 +158,7 @@ class ClearlyDefinedStorage(
                 loadScanCodeResults(id, coordinates, version, startTime)
             } ?: emptyResult(id)
         } catch (e: IOException) {
-            log.error("Error when reading results for package '${id.toCoordinates()}' from ClearlyDefined.", e)
+            log.error { "Error when reading results for package '${id.toCoordinates()}' from ClearlyDefined." }
```
Comment (mnonnenmacher) [inline scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt:161]: You should also add `e.showStackTrace()`.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt (hunk 1 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt:130]: I decided to only add the call to `e.showStackTrace()` as seeing the exception message as part of the log was exactly what [confused a user](https://github.com/oss-review-toolkit/ort/issues/3239).

Comment (mnonnenmacher) [reply scanner/src/main/kotlin/storages/ClearlyDefinedStorage.kt:132]: We use this order and formatting everywhere else in the code:
```
...catch(e: Exception) {
  e.showStackTrace()

  log.warn { ... }

  ...
}
```

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

**Comment by sschuberth** [comment]
> We use this order and formatting everywhere else in the code:

Done.

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**